src/context/Context.js

```diff
@@ -0,0 +1,65 @@
+import React, { useContext, useEffect, useReducer } from 'react';
+import reducer from '../reducer/Reducer';
+
+const AppContext = React.createContext();
+
+const getProductsFromLs = (_) => {
+    const lsProducts = JSON.parse(localStorage.getItem('product'));
+    return localStorage.getItem('product') !== null ? lsProducts : [];
+};
+
+const initialState = {
+    isOpen: false,
+    cart: getProductsFromLs(),
+    total: 0,
+    amount: 0,
+};
```

Comment:
Ovo je sve dobro samo sam hteo da znas da reducer ne moras da koristis u contextu ako je logika jednostavna. Ti si mogao umesto svega ovog da imas 4 lokalna state-a i da provajderu samo exportujes isOpen, setIsOpen, cart, setCart, total, setTotal...

Ne kazem da si to trabao vec samo da imas i tu opciju ako zatreba...